boards/arduino-mega2560/src/lib.rs

```diff
@@ -22,7 +22,17 @@ pub use crate::hal::spi;
 pub use crate::hal::adc;
 
 pub type Delay = crate::hal::delay::Delay<hal::clock::MHz16>;
-pub type Serial<IMODE> = crate::hal::usart::Usart0<hal::clock::MHz16, IMODE>;
```

Comment:
I'd really keep this alias and just change it to

```rust
pub type Serial<IMODE> = crate::usart::Usart0<IMODE>;
```

then you also don't need to update all the examples.

The `Serial` is supposed to be the one that is directly connected to the USB-serial-converter, i.e. the one you'll interact with from your PC.  This one is made more accessible than the others because its the one that will be used the most.